src/figuresClasses.ts

```diff
@@ -1,11 +1,89 @@
-export interface Figure {}
+type Color = `red` | `green` | `blue`;
 
-export class Triangle implements Figure {}
+type Shape = `triangle` | `circle` | `rectangle`;
 
-export class Circle implements Figure {}
+interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  sides: number[];
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('Invalid triangle sides: cannot form a triangle');
+    }
+    this.sides = [this.a, this.b, this.c];
+  }
+
+  getArea(): number {
+    const [a, b, c] = this.sides;
```

Comment:
You still use `this.sides`. The problem is that you don't have one source of truth as you have stored values `a, b, c` and in the array sides. So if any value changes you have to update in two places